### PR TITLE
Disable cpu-only-nodes preferred scheduling that breaks placement groups.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

The attached regression test (which simulates the critical user's workload) is broken when `scheduler_avoid_gpu_nodes ` is true. It seems like the gpu scheduling has a bug when placement group is used. The PR itself remedies the issue by disabling this feature, and I will open issues for existing issues with a local repro. 

The PR also adds the regression placement group test that's running in an autoscaling cluster. This test is probably not necessary as my local cluster setup seems to be able to reproduce the issue. 

As a follow up, I created these 2 issues
https://github.com/ray-project/ray/issues/19131 -> placement group resource leak when this issue happens
https://github.com/ray-project/ray/issues/19130 -> reenable this feature

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/placement_group_tests/cluster.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from ray.cluster_utils import Cluster
```
Comment (rkooo567) [inline release/nightly_tests/placement_group_tests/cluster.py:1]: This file is needed for the local repro. I will make issues for local repro

File: release/nightly_tests/placement_group_tests/compute.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      max_workers: 2
+      use_spot: false
+    - name: fake_gpu_node
+      instance_type: m5.4xlarge
+      min_workers: 0 
+      max_workers: 2
+      use_spot: false
+      resources:
+        cpu: 16
+        gpu: 1
```
Comment (rkooo567) [inline release/nightly_tests/placement_group_tests/compute.yaml:27]: we simulate gpu nodes to save cost

File: release/nightly_tests/placement_group_tests/pg_run.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    ray.init(address="auto")
+
+    bundles = [{"CPU": 1, "GPU": 1}]
+    bundles += [{"CPU": 1} for _ in range(30)]
+
+    pg = placement_group(bundles, strategy="PACK")
+
+    ray.get(pg.ready())
+
+    workers = [Worker.options(placement_group=pg).remote(i) for i in range(30)]
```
Comment (gjoliver) [inline release/nightly_tests/placement_group_tests/pg_run.py:42]: maybe extract 30 into a constant. NUM_WORKERS

**Review by gjoliver — APPROVED** [review]

File: release/nightly_tests/placement_group_tests/cluster.py (hunk 1 above)
Comment (sasha-s) [reply release/nightly_tests/placement_group_tests/cluster.py:1]: Thank you for adding the repro!

**Pull request merged**